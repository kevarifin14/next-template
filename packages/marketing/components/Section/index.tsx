import { ReactNode } from 'react';
import { classNames, TailwindSize } from 'shared';

type SectionProps = {
  children: ReactNode,
  size?: TailwindSize,
  className?: string,
};

export default function Section({ children, size, className }: SectionProps) {
  const sectionClassName = classNames(
    'py-8 px-4 lg:py-12 lg:px-6 w-full prose dark:prose-light mx-auto',
    size ? `max-w-${size}` : 'max-w-none',
    className,
  );

  return (
    <div className={sectionClassName}>
      {children}
    </div>
  );
}
