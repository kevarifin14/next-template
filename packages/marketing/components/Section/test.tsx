import { render } from '@testing-library/react';

import Section from '.';

describe('Section', () => {
  it('matches snapshot', () => {
    const { baseElement } = render(<Section>mock</Section>);
    expect(baseElement).toMatchSnapshot();
  });
});
