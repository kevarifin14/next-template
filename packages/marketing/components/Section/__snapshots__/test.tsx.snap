// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section matches snapshot 1`] = `
<body>
  <div>
    <div
      class="py-8 px-4 lg:py-12 lg:px-6 w-full prose dark:prose-light mx-auto max-w-none"
    >
      mock
    </div>
  </div>
</body>
`;
